assets/src/components/propertiesPanel/minischedule.tsx

```diff
@@ -126,28 +142,51 @@ const DeadheadTrip = ({
 }) => {
   const startTime: string = formattedScheduledTime(trip.startTime)
   if (sequence === "first") {
-    return (
-      <Row icon={questionMarkIcon()} text={"Pull Out"} rightText={startTime} />
-    )
+    return <Row icon={busFrontIcon()} text={"Pull Out"} rightText={startTime} />
   } else if (sequence === "last") {
-    return (
-      <Row icon={questionMarkIcon()} text={"Pull Back"} rightText={startTime} />
-    )
+    return <Row icon={busRearIcon()} text={"Pull Back"} rightText={startTime} />
   } else {
     return (
-      <Row icon={questionMarkIcon()} text={"Deadhead"} rightText={startTime} />
+      <Row icon={filledCircleIcon()} text={"Deadhead"} rightText={startTime} />
     )
   }
 }
 
+const iconForDirectionOnLadder: (
+  directionId: DirectionId,
+  ladderDirections: LadderDirections,
+  routeId: RouteId
+) => ReactElement = (directionId, ladderDirections, routeId) => {
+  const ladderDirection = getLadderDirectionForRoute(ladderDirections, routeId)
+  if (
+    directionOnLadder(directionId, ladderDirection) === VehicleDirection.Down
+  ) {
+    return triangleDownIcon()
+  }
+  return triangleUpIcon()
+}
+
 const RevenueTrip = ({ trip }: { trip: Trip }) => {
   const startTime: string = formattedScheduledTime(trip.startTime)
   const formattedVariant: string =
     trip.viaVariant !== null && trip.viaVariant !== "_" ? trip.viaVariant : ""
   const formattedRouteAndVariant: string = `${trip.routeId}_${formattedVariant}`
+  const [{ ladderDirections }] = useContext(StateDispatchContext)
+
+  const directionIcon =
+    trip.directionId !== null
+      ? // Safe to assume routeId is not null, since if it were, we'd be
+        // rendering a deadhead row instead.
+        iconForDirectionOnLadder(
+          trip.directionId,
+          ladderDirections,
+          trip.routeId!
+        )
+      : questionMarkIcon()
```

Comment:
If `iconForDirectionOnLadder` takes `DirectionId | null`, then all the decision about which icon to show can happen in the one function, instead of being split between that function and this `?:`.